### PR TITLE
Add DotEditor.app v0.3.1

### PR DESCRIPTION
**Pull request by shageman**
This PR adds DotEditor, a graphviz dotfile editor.

After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not already refused in [closed issues].
- [X] Checked the cask is submitted to [the correct repo].

### CODE REVIEW

File: Casks/doteditor.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  url "https://github.com/vincenthEE/DotEditor/releases/download/v#{version}/DotEditor.#{version}.dmg"
+  appcast 'https://github.com/vincenthEE/DotEditor/releases.atom',
+          checkpoint: 'aeffe7fe003602a870319b4494444b6d97898d63047c330138011487a5c37a21'
+  name 'DotEditor'
+  homepage 'http://vincenthee.github.io/DotEditor/'
+
+  app 'DotEditor.app'
+
+  caveats "Graphviz needs to be installed for #{token} to work."
```
Comment (vitorgalvao) [inline Casks/doteditor.rb]: Use [`depends_on`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/depends_on.md) instead.

File: Casks/doteditor.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+cask 'doteditor' do
+  version '0.3.1'
+  sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/doteditor.rb]: This needs to be set. Run `shasum -a 256` on the downloaded file.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/doteditor.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  url "https://github.com/vincenthEE/DotEditor/releases/download/v#{version}/DotEditor.#{version}.dmg"
```
Comment (vitorgalvao) [inline Casks/doteditor.rb]: This needs a comment above, that the `style` command will have warned about.

**Review by vitorgalvao — APPROVED** [review]

**Comment by victorpopkov** [comment]
Thank you for your contribution!

**Pull request merged**